ci_framework/plugins/modules/get_makefiles_env.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2023, Chandan Kumar <chkumar@redhat.com>
+# Apache License Version 2.0 (see LICENSE)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: get_makefiles_env
+
+short_description: Recursively get Makefiles default variables.
+
+description:
+- Retrieves a dictionary with all the environment variables
+  defined in the Makefiles under the given path.
+
+options:
+  base_path:
+      description:
+      - The base path on where to start picking Makefiles vars.
+      required: True
+      type: str
+author:
+  - Chandan Kumar (@raukadah)
+  - Cedric Jeanneret (@cjeanner)
+  - Pablo Rodriguez (@pablintino)
+'''
+
+EXAMPLES = r'''
+# Recursively get all the variables of the Makefiles under /home/user
+- name: Generate env var fact
+  get_makefiles_env:
```

Comment:
I think we should add "register: my_makefile_params" in order to emphasis it won't end in the facts.